packages/firebase_storage/firebase_storage_web/lib/src/firebase_storage_web.dart

```diff
@@ -21,44 +20,43 @@ typedef ReferenceBuilder = ReferencePlatform Function(
 
 /// The Web implementation of the FirebaseStoragePlatform.
 class FirebaseStorageWeb extends FirebaseStoragePlatform {
-  /// The js-interop layer for Firebase Storage
-  final storage_interop.Storage webStorage;
-
-  // Same default as the method channel implementation
-  int _maxDownloadRetryTime = Duration(minutes: 10).inMilliseconds;
-
-  // Same default as the method channel implementation
-  int _maxOperationRetryTime = Duration(minutes: 2).inMilliseconds;
-
-  // Empty constructor. This is only used by the registerWith method.
-  FirebaseStorageWeb._nullInstance() : webStorage = null;
-
   /// Construct the plugin.
   /// (Web doesn't use the `bucket`, since the init happens in index.html)
-  FirebaseStorageWeb({FirebaseApp app, String bucket})
+  FirebaseStorageWeb({FirebaseApp? app, required String bucket})
       : webStorage =
             storage_interop.getStorageInstance(core_interop.app(app?.name)),
         super(appInstance: app, bucket: bucket);
 
+  // Empty constructor. This is only used by the registerWith method.
+  // superclass also needs to be initialized and 'bucket' param is required.
+  FirebaseStorageWeb._nullInstance()
+      : webStorage = null,
+        super(bucket: '');
+
   /// Create a FirebaseStorageWeb injecting a [fb.Storage] object.
   @visibleForTesting
-  FirebaseStorageWeb.forMock(this.webStorage, {String bucket, FirebaseApp app})
+  FirebaseStorageWeb.forMock(this.webStorage,
+      {required String bucket, FirebaseApp? app})
       : super(appInstance: app, bucket: bucket);
 
+  /// The js-interop layer for Firebase Storage
+  final storage_interop.Storage? webStorage;
```

Comment:
This is nullable. This is the same as auth web. although it violate that principle you mentioned yesterday, @Salakar. It's only nullable to satisfy  `registerWith`. I wonder if it is necessary.